pkg/api/node.go

```diff
@@ -92,9 +91,8 @@ func (m *nodeMetrics) List(ctx context.Context, options *metainternalversion.Lis
 
 	metricsItems, err := m.getNodeMetrics(nodes...)
 	if err != nil {
-		errMsg := fmt.Errorf("Error while fetching node metrics for selector %v: %v", labelSelector, err)
-		klog.Error(errMsg)
-		return &metrics.NodeMetricsList{}, errMsg
+		klog.ErrorS(err, "Failed getting nodes metrics", "labelSelector", labelSelector)
```

Comment:
nit: For List() we say "Failed getting" whereas for Get() we say "Failed reading". To make things clearer, we might want to keep the same wording for both cases.
From what you did for pods, I'd say we should change for "reading" here.

Reply:
done